File: src/dataset/cpp/bob_exercism-Richard-m-j/openAI/ChatGPT4_bob_v2.cpp
#include "bob.h"
#include <string>
#include <algorithm>
#include <cctype>
using namespace std;

namespace bob
{
    string hey(string input)
    {
        // Trim whitespace from both ends
        input.erase(input.begin(), find_if(input.begin(), input.end(), [](unsigned char ch) { return !isspace(ch); }));
        input.erase(find_if(input.rbegin(), input.rend(), [](unsigned char ch) { return !isspace(ch); }).base(), input.end());

        if (input.empty())
            return "Fine. Be that way!";
        if (isYelling(input) && input.back() == '?')
            return "Calm down, I know what I'm doing!";
        if (isYelling(input))
            return "Whoa, chill out!";
        if (input.back() == '?')
            return "Sure.";
        return "Whatever.";
    }

    bool isYelling(const string &input)
    {
        bool hasLetters = false;
        for (char ch : input)
        {
            if (isalpha(ch))
            {
                hasLetters = true;
                if (islower(ch))
                    return false;
            }
        }
        return hasLetters;
    }

} // namespace bob

File: src/dataset/typescript/bob_exercism-samajammin/openAI/ChatGPT4_bob_v2.ts
export default class Bob {
  hey(message: string): string {
    const trimmedMessage = message.trim();
    if (!trimmedMessage) return 'Fine. Be that way!';
    const isYelling = /^[^a-z]*[A-Z][^a-z]*$/.test(trimmedMessage);
    if (isYelling) return 'Whoa, chill out!';
    if (trimmedMessage.endsWith('?')) return 'Sure.';
    return 'Whatever.';
  }
}